crates/bevy_pbr/src/material.rs

```diff
@@ -1471,13 +1494,138 @@ impl<M: Material> RenderAsset for PreparedMaterial<M> {
             _ => None,
         };
 
-        match material.unprepared_bind_group(
-            &pipeline.material_layout,
-            render_device,
-            material_param,
-            false,
-        ) {
+        let mut draw_functions = HashMap::new();
+        draw_functions.insert(MaterialDrawFunction.intern(), draw_function_id);
+        if let Some(prepass_draw_function_id) = prepass_draw_function_id {
+            draw_functions.insert(PrepassDrawFunction.intern(), prepass_draw_function_id);
+        }
+        if let Some(deferred_draw_function_id) = deferred_draw_function_id {
+            draw_functions.insert(DeferredDrawFunction.intern(), deferred_draw_function_id);
+        }
+        if let Some(shadow_draw_function_id) = shadow_draw_function_id {
+            draw_functions.insert(ShadowsDrawFunction.intern(), shadow_draw_function_id);
+        }
+
+        let mut shaders = HashMap::new();
+
+        let vertex_shader = match M::vertex_shader() {
+            ShaderRef::Default => None,
+            ShaderRef::Handle(handle) => Some(handle),
+            ShaderRef::Path(path) => Some(asset_server.load(path)),
+        };
```

Comment:
This should probably be added as a utility function on `ShaderRef`. Like `ShaderRef::materialize(&self, asset_server) -> Option<Handle<Shader>>`

Reply:
Not sure about the name, that's just the first thing I could think of.